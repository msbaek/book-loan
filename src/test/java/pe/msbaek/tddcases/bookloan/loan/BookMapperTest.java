package pe.msbaek.tddcases.bookloan.loan;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.modulith.test.ApplicationModuleTest;

import java.time.LocalDate;

@ApplicationModuleTest
class BookMapperTest {
    @Autowired BookMapper bookMapper;

    @Test
    void toDto() {
        // given
        Book book = new Book("title", "author", LocalDate.now());

        // when
        BookDto dto = bookMapper.toDto(book);
        Approvals.verify(dto);
    }

    @Test
    void toEntity() {
        // given
        BookDto dto = new BookDto(1L, "title", "author", "2021-08-01");

        // when
        Book book = bookMapper.toEntity(dto);
        Approvals.verify(book.toString());
    }
}