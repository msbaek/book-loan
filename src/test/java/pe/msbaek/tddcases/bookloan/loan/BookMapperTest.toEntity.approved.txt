Book{id=1, title='title', author='author', publishedDate=2021-08-01, copies=[]}